### PR TITLE
Normalize and Update Package Names in Pipfiles Using PyPI Data

### PR DESCRIPTION
**Pull request by devcoons**
This PR adds a script that recursively scans a specified directory for all Pipfiles, queries PyPI to retrieve the canonical package names, and normalizes them. It updates the package names in-place to ensure consistent casing and format across all Pipfiles.

## Summary by CodeRabbit

* **New Features**
  * Introduced a script to automatically update and normalize package names in Pipfiles based on canonical PyPI naming conventions.
* **Bug Fixes**
  * Corrected package names in multiple Pipfiles to match their official PyPI identifiers (e.g., "MarkupSafe" to "markupsafe", "tensorflow_rocm" to "tensorflow-rocm").
  * Updated versions and hashes for several Python packages including aiohttp, comm, contourpy, gitpython, grpcio, keras, ml-dtypes, narwhals, opentelemetry packages, optree, regex, tokenizers, and huggingface-hub across various environments.
* **Documentation**
  * Added a usage example for the new package name normalization script in the documentation.

### CODE REVIEW

File: jupyter/tensorflow/ubi9-python-3.11/Pipfile [hunk 1]
```diff
@@ -37,7 +37,7 @@ mysql-connector-python = "~=9.3.0"
 odh-elyra = "==4.2.2"
 
 jupyterlab = "==4.2.7"
-jupyter-bokeh = "~=4.0.5"
+jupyter_bokeh = "~=4.0.5"
```
[the remark this answers is not in the document]
Comment (devcoons) [reply jupyter/tensorflow/ubi9-python-3.11/Pipfile:40]: According to PyPi the correct package name is with underscore:
https://pypi.org/project/jupyter_bokeh/2.0.2/

File: jupyter/rocm/pytorch/ubi9-python-3.12/Pipfile [hunk 2]
```diff
@@ -44,7 +44,7 @@ mysql-connector-python = "~=9.3.0"
 odh-elyra = "==4.2.2"
 
 jupyterlab = "==4.2.7"
-jupyter-bokeh = "~=4.0.5"
+jupyter_bokeh = "~=4.0.5"
```
[the remark this answers is not in the document]
Comment (devcoons) [reply jupyter/rocm/pytorch/ubi9-python-3.12/Pipfile:47]: According to PyPi the correct package name is with underscore:
https://pypi.org/project/jupyter_bokeh/2.0.2/

File: jupyter/datascience/ubi9-python-3.12/Pipfile [hunk 3]
```diff
@@ -32,7 +32,7 @@ mysql-connector-python = "~=9.2.0"
 odh-elyra = "==4.2.2"
 
 jupyterlab = "==4.2.7"
-jupyter-bokeh = "~=4.0.5"
+jupyter_bokeh = "~=4.0.5"
```
[the remark this answers is not in the document]
Comment (jiridanek) [reply jupyter/datascience/ubi9-python-3.12/Pipfile:35]: @coderabbitai link something I can read up on this canonical name idea

[the remark this answers is not in the document]
Comment (jiridanek) [reply jupyter/datascience/ubi9-python-3.12/Pipfile:35]: > So jupyter_bokeh, jupyter-bokeh, Jupyter.Bokeh, and JUPYTER_BOKEH all normalize to: jupyter-bokeh

Why is this PR wanting to switch us to `jupyter_bokeh`, then?

File: scripts/fix_package_naming.py
[the remark this answers is not in the document]
Comment (atheo89) [reply scripts/fix_package_naming.py:1]: I found the original version to be cleaner and easier to understand. CodeRabbit essentially removed the comments and compressed the lines, making the code more difficult to read.

**Comment by atheo89** [comment]
Thank you for your contribution @devcoons! It’s a significant improvement that resolves many naming issues in the Pipfiles. I double-checked the changes on PyPI, and they align well with the canonical package names.
Much appreciated!

/lgtm
/approve
